cpp/service.cc

```diff
@@ -1,13 +1,15 @@
 #include "service.h"
 
 ServiceLayer::ServiceLayer(KeyValueClientInterface* key_value_connection) {
+  service_mtx_ = new std::mutex();
```

Comment:
Generally you want to make a mutex a non-pointer member variable and then just initialize it in the initializer list.